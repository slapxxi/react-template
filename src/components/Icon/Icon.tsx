import * as React from 'react';

interface Props {
  glyph: Glyph;
  size?: number;
  className?: string;
}

function Icon({ className, glyph, size = 100 }: Props) {
  return (
    <svg
      width={size}
      height={size}
      viewBox={glyph.viewBox}
      className={className}
    >
      <use xlinkHref={`#${glyph.id}`} />
    </svg>
  );
}

export default Icon;
