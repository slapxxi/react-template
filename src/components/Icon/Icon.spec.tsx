import React from 'react';
import { render } from 'react-testing-library';
import Icon from './Icon';

it('renders icon with specified name', () => {
  const { container } = render(
    <Icon glyph={{ id: 'icon', viewBox: '0 0 10 10' }} />,
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts className', () => {
  const { container } = render(
    <Icon glyph={{ id: 'icon', viewBox: '0 0 10 10' }} className="icon" />,
  );
  expect(container.firstChild).toMatchSnapshot();
});
