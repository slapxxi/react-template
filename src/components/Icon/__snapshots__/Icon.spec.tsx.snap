// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts className 1`] = `
<svg
  class="icon"
  height="100"
  viewBox="0 0 10 10"
  width="100"
>
  <use
    xlink:href="#icon"
  />
</svg>
`;

exports[`renders icon with specified name 1`] = `
<svg
  height="100"
  viewBox="0 0 10 10"
  width="100"
>
  <use
    xlink:href="#icon"
  />
</svg>
`;
